### PR TITLE
SwG PoC: Gutenberg plugin

### PR DESCRIPTION
**Pull request by ChrisAntaki**
## Summary

Addresses issue #

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { Component } from '@wordpress/element';
+
+export default class AccessSelector extends Component {
+	constructor( props ) {
+		super( props );
+
+		const { select } = global.wp.data;
+		const postMeta = select( 'core/editor' ).getEditedPostAttribute(
+			'meta'
+		);
```
Comment (tofumatt) [inline assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:32]: This should be done with the [`withSelect` higher-order component](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#withselect), or this data will not update when the data store does.

File: assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		this.state = {
+			access: postMeta.sitekit__reader_revenue__access,
+		};
+	}
+
+	setAccess( access ) {
+		// Update post meta field.
+		const { dispatch } = global.wp.data;
```
Comment (tofumatt) [inline assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:40]: Similarly, this should use [`withDispatch`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#withdispatch) and not the global dispatch.

**Comment by eugene-manuilov** [comment]
@ChrisAntaki this might be helpful to you: https://developer.wordpress.org/block-editor/how-to-guides/metabox/

File: assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js (hunk 1 above)
Comment (ChrisAntaki) [reply assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js]: Good call, updated!

File: assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	withDispatch( ( dispatch ) => ( {
+		setAccess: ( access ) =>
+			dispatch( 'core/editor' ).editPost( {
+				meta: { sitekit__reader_revenue__access: access },
+			} ),
+	} ) ),
+	withSelect( ( select ) => ( {
+		access: select( 'core/editor' ).getEditedPostAttribute( 'meta' )
+			.sitekit__reader_revenue__access,
+		products: global.googlesitekit.data.select( STORE_NAME ).getProducts(),
```
Comment (ChrisAntaki) [inline assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:91]: Hey @tofumatt ! Just wondering, do we have a good solution for using Site Kit's `withSelect` for a Gutenberg component? I tried to use Site Kit's `withSelect` alongside WordPress', in this call to `compose`, but then WordPress crashed and mentioned something about two versions of React. What do you think?

Comment (ChrisAntaki) [reply assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:91]: cc @aaemnnosttv and @felixarntz

**Comment by tonyjemba** [comment]
checks still failing

File: assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js (hunk 3 above)
Comment (aaemnnosttv) [reply assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:91]: >  I tried to use Site Kit's withSelect alongside WordPress', in this call to compose, but then WordPress crashed and mentioned something about two versions of React. What do you think?

Hey @ChrisAntaki, not sure if this is still an issue for you but the reason this is due to `withSelect` using hooks internally which requires that the same version of React is used and Site Kit's will always reference the version of React we bundle. So when using within the context of Gutenberg, we'll always need to use `withSelect` via the `wp` global for the same reason we need to use class-based components here.

If we don't already have an issue for using the two data stores together we'll need to create one for that.

Comment (ChrisAntaki) [reply assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:91]: Thanks for the explanation, @aaemnnosttv !

Hey @felixarntz , just checking, did you already create an issue for this a while back ?

Comment (felixarntz) [reply assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:91]: @aaemnnosttv I created #4107 (and related #4108 + #4109) around this, but it doesn't actually encompass the ability to use stores from both registries in a single component yet I believe. Can you check where the new issue fits in with those?

Comment (aaemnnosttv) [reply assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js:91]: @felixarntz @ChrisAntaki I opened https://github.com/google/site-kit-wp/issues/4357. With this we would use the Site Kit registry for accessing both our stores and WP core stores – we would still need to use the HOCs via `wp` though as mentioned before.

**Comment by ChrisAntaki** [comment]
This issue causes a JS test failure: https://github.com/google/site-kit-wp/issues/4831

**Pull request closed without merging**